src/js/heuristicblocking.js

```diff
@@ -125,12 +120,125 @@ HeuristicBlocker.prototype = {
       return {};
     }
 
-    // ignore if there are no tracking cookies
-    if (!hasCookieTracking(details, origin)) {
-      return {};
+    // check if there are tracking cookies
+    if (hasCookieTracking(details, origin)) {
+      this._recordPrevalence(fqdn, origin, tabOrigin);
     }
+  },
 
-    this._recordPrevalence(fqdn, origin, tabOrigin);
+  pixelCookieShareListener: function (details) {
+    // check if this request is a cookie-syncing pixel
+    if (details.type == 'image') {
+      let self = this,
+        frameHost = badger.tabData[details.tabId].frames[details.frameId].host;
+
+      // get all cookies for the frame the request comes from and pass those to
+      // the cookie-share accounting function
+      chrome.cookies.getAll({
+        domain: window.getBaseDomain(frameHost)
```

Comment:
Is the getBaseDomain() necessary? If it is, could we avoid it by using the `url` filter instead?